platform/linux-generic/odp_schedule_sp.c

```diff
@@ -107,7 +108,7 @@ typedef struct ODP_ALIGNED_CACHE sched_group_t {
 
 		/* All groups */
 		struct {
-			char          name[ODP_SCHED_GROUP_NAME_LEN + 1];
+			char          name[ODP_SCHED_GROUP_NAME_LEN];
```

Comment:
This is a functional change that shortens the maximum group name length. It was a bit unexpected as it was not mentioned in the commit message.

The API spec says in schedule.h that "maximum string length is ODP_SCHED_GROUP_NAME_LEN", i.e. excluding the terminating null character, but in schedule_types.h it says that ODP_SCHED_GROUP_NAME_LEN includes the null character. This is inconsistent and probably should be fixed in the API. ODP_TIMER_POOL_NAME_LEN has the same problem.

So, since the API is inconsistent, the old behavior seems more correct since it works with either interpretation of the constant. However, the other schedulers already include the null character in the group name length, so maybe it is ok to make the sp work the same way.

Reply:
This was hidden because I didn't even consider it an inconsistency, but more like an inaccuracy. Either way, I agree that the API should be clarified.

This change is a separate commit in v2.